.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,59 @@
+name: Benchmarking tests
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - ci-*
+
+env:
+  DIFFUSERS_IS_CI: yes
+  HF_HOME: /mnt/cache
+  OMP_NUM_THREADS: 8
+  MKL_NUM_THREADS: 8
+  PYTEST_TIMEOUT: 600
+  RUN_SLOW: yes
+  PIPELINE_USAGE_CUTOFF: 50000
+
+jobs:
+  torch_pipelines_cuda_benchmark_tests:
+    name: Torch Core Pipelines CUDA Benchmarking Tests
```

Comment:
Depending on how you're defining core, you could use this script to fetch the relevant pipeline modules for testing
https://github.com/huggingface/diffusers/blob/main/utils/fetch_torch_cuda_pipeline_test_matrix.py

Reply:
Great idea, but there can be some pipeline-specific parameters in the benchmarking script (for example the checkpoint id, image inputs, etc.). So, I would prefer to just have isolated benchmarking scripts sharing common parameters and functions. 

This way, the scripts themselves stay minimal and clean. They also become quite easy to debug and serve as reference pointers. 

WDYT about this approach?